### PR TITLE
feat(test): redesign the way of Net communicate with peers

### PR DESCRIPTION
**Pull request by keroro520**
Based on https://github.com/nervosnetwork/ckb/pull/2286. So please review https://github.com/nervosnetwork/ckb/pull/2286  first.

There are 2 significant updations:

  * Initialize `Net`'s network controller when creating `Net` Instance.
  * Make the communication channels peer-level. Which is say, we can receive network messages from specific peers. This feature prevents peers' message confusion.

### CODE REVIEW

File: test/src/net.rs [hunk 1]
```diff
@@ -57,129 +96,151 @@ impl Net {
...
+
+    pub fn connect(&mut self, node: &Node) {
+        self.controller().add_node(
+            &node.node_id().parse().unwrap(),
+            node.p2p_address().parse().unwrap(),
+        );
+
+        let mut n_protocols_connected = 0;
+        while let Ok(connected) = self.register_rx.recv_timeout(Duration::from_secs(60)) {
+            log::info!("bilibili");
```
Comment (quake) [inline test/src/net.rs:114]: bilibili ? LOL

**Review by quake — DISMISSED** [review]

**Review by quake — APPROVED** [review]

**Review by chuijiaolianying — APPROVED** [review]

**Comment by chuijiaolianying** [comment]
bors r=quake,chuijiaolianying

**Comment by doitian** [comment]
bors retry

**Comment by keroro520** [comment]
bors retry

**Pull request merged**